src/wgan-gp.jl

```diff
@@ -3,25 +3,29 @@ Zygote.@adjoint CUDA.ones(x...) = CUDA.ones(x...), _ -> map(_ -> nothing, x)
 Zygote.@adjoint CUDA.rand(x...) = CUDA.rand(x...), _ -> map(_ -> nothing, x)
 
 """
-    lipschitz1_gradient_loss(m, x_true, x_generated, batch_size)
+    lipschitz1_gradient_loss(m, x_true, x_generated)
 Estimates 𝐄ₓ(‖∇ₓD(x)‖₂ - 1)², where x is sampled uniformly on lines between
 points from the data distribution and the generators distribution
 """
-function lipschitz1_gradient_loss(m, x_true, x_generated, batch_size)
-    ξ = CUDA.rand(1, 1, 1, batch_size)
+function lipschitz1_gradient_loss(m, x_true, x_generated)
+    x_size = size(x_true)
+    batch_size = x_size[end]
+    data_dims = length(x_size) - 1
+    ξ_size = tuple(ones(Int,data_dims)..., batch_size)
+    ξ = CUDA.rand(ξ_size...)
     x_interpolated = ξ .* x_true + (1.0f0 .- ξ) .* x_generated
     _, b = pullback(() -> m(x_interpolated), params(x_interpolated))
     grads = b(CUDA.ones(1, batch_size))
     sqddx = grads[x_interpolated] .^ 2
-    mean((sqrt.(sum(sqddx, dims = (1, 2))) .- 1.0f0) .^ 2)
+    mean((sqrt.(sum(sqddx, dims = (1:data_dims))) .- 1.0f0) .^ 2)
```

Comment:
```suggestion
    mean((sqrt.(sum(sqddx, dims = (1:data_dims...,))) .- 1.0f0) .^ 2)
```
not sure, but this would be more like (1,2)

Reply:
seems to work either way